### PR TITLE
Create advanced-alarm-handling.adoc

### PR DESCRIPTION
**Pull request by marshallmassengill**
Adding a page for examples of advanced alarm handling with drools.

### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/${JIRA-ISSUE-NUMBER}

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+[[ga-Included-alarm-handling]]
+== Included Alarm handling
+
+In addition to the manual actions, it is possible to automate alarm handling with the use of https://www.drools.org/[Drools] scripts.
+
+Within the `$\{OPENNMS_HOME}/etc/alarmd/drools-rules.d/alarmd.drl` file there is a default rule set of rules for handling alarm cleanup, 
+clearing, and creating/updating tickets.
+
+Within the Within the `$\{OPENNMS_HOME}/etc/alarmd/drools-rules.d/situations.drl` file there is a default rule set for handling alarm cleanup, 
+clearing, and creating/updating tickets.
```
Comment (dino2gnt) [inline docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc:15]: Wasn't Within the Within a crappy 90's prog band?
```suggestion
The `$\{OPENNMS_HOME}/etc/alarmd/drools-rules.d/situations.drl` file contains the default rules for the Situation lifecycle.
```

File: docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+You can find some additional examples in the `$\{OPENNMS_HOME}/etc/examples/alarmd/drools-rules.d/` directory.
```
Comment (dino2gnt) [inline docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc:17]:
```suggestion
Additional examples are available in `$\{OPENNMS_HOME}/etc/examples/alarmd/drools-rules.d/`.
```

**Review by dino2gnt — CHANGES_REQUESTED** [review]
It's a great start.

File: docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc (hunk 1 above)
Comment (dino2gnt) [reply docs/modules/operation/pages/deep-dive/alarms/advanced-alarm-handling.adoc]: Also, no mid-sentence line breaks (as I recall).  Keep everything on one line.

**Comment by dino2gnt** [comment]
What is this `master-33` branch this is targeting?  That should be ... something else, I think.

**Comment by marshallmassengill** [comment]
> What is this `master-33` branch this is targeting?  That should be ... something else, I think.

Ugh... yeah... will resolve that tomorrow.

**Review by dino2gnt — APPROVED** [review]

**Review by cgorantla — APPROVED** [review]

**Comment by cgorantla** [comment]
@marshallmassengill  you may have to merge latest develop into your branch to make the circleci pass the builds

**Comment by marshallmassengill** [comment]
> @marshallmassengill you may have to merge latest develop into your branch to make the circleci pass the builds

I think I've done that?

**Review by indigo423 — APPROVED** [review]
LGTM

**Comment by indigo423** [comment]
@marshallmassengill you are aware this change will end up in Horizon 34.x. If you want to have this also in Meridian releases you need to base your contribution on the earliest Meridian branch your description will work.

**Comment by marshallmassengill** [comment]
Not worried about this one being in the Meridian docs though I guess it could be.  Let's get it added first and we can go from there.

**Pull request merged**